Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -427,60 +421,134 @@ function Unregister-ServiceFabricApplicationTypeAction
         $TimeoutSec
     )
 
-    $global:operationId = $SF_Operations.UnregisterApplicationType
+    $global:operationId = $SF_Operations.GetApplicationType
+    $getAppTypeAction = { Get-ServiceFabricApplicationTypeAction -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion }
+    $getAppTypeRetryEvaluator = {
+        param($appType)
 
-    $unregisterAction = {
-        Unregister-ServiceFabricApplicationType -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion -Force -TimeoutSec $TimeoutSec
-        if (!$?)
+        # If provisioning not started, retry register
+        if(!$appType)
         {
-            throw (Get-VstsLocString -Key SFSDK_UnableToUnregisterAppType)
+            return $true
         }
-    }
-
-    $exceptionRetryEvaluator = {
-        param($ex)
-        $appType = Get-ServiceFabricApplicationTypeAction -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion
-        # If app type already unprovisioned, don't retry
-        if (!$appType)
+        # if app type is provisioned, don't retry
+        elseif($appType -and ($appType.Status -eq [System.Fabric.Query.ApplicationTypeStatus]::Available))
```

Comment:
You don't need to put $appType check here since it would not come here appType is null

Reply:
Correct